src/TableFunctions/TableFunctionObjectStorage.cpp

```diff
@@ -100,27 +112,53 @@ StoragePtr TableFunctionObjectStorage<Definition, Configuration>::executeImpl(
     ColumnsDescription cached_columns,
     bool is_insert_query) const
 {
-    ColumnsDescription columns;
     chassert(configuration);
+    ColumnsDescription columns;
+
     if (configuration->structure != "auto")
         columns = parseColumnsListFromString(configuration->structure, context);
     else if (!structure_hint.empty())
         columns = structure_hint;
     else if (!cached_columns.empty())
         columns = cached_columns;
 
-    StoragePtr storage = std::make_shared<StorageObjectStorage>(
+    StoragePtr storage;
+    const auto & settings = context->getSettingsRef();
+
+    auto parallel_replicas_cluster_name = settings[Setting::cluster_for_parallel_replicas].toString();
+    auto can_use_parallel_replicas = settings[Setting::allow_experimental_parallel_reading_from_replicas] > 0
```

Comment:
Instead of direct parallel replicas settings check, prefer to use https://github.com/ClickHouse/ClickHouse/blob/dd092411ffe9da25d190a0089396b4642ee31cdb/src/Interpreters/Context.cpp#L5882-L5889

Reply:
Saw this [comment](https://github.com/ClickHouse/ClickHouse/pull/70659#discussion_r1850033123). Well, then the comment is not relevant